### PR TITLE
Cross-build plugin for SBT 1.0

### PR DESCRIPTION
**Pull request by rtimush**
The plugin can be now built both for SBT 0.13 and 1.0 series, use `^^task` to cross-build or `^1.0.1` to switch SBT version.

There was some incompatibility between SBT 0.13 and 1.0 related to `CrossVersionUtil` easily fixed by a thin compatibility layer. The bigger issue was cloning of a project with a new id — there is no `project.copy(id = ...)` in SBT 1.0. I worked around by creating a fresh project and applying all available configuration options from the original one, but haven't thoroughly tested this.

### CODE REVIEW

**Review by vi-kas — APPROVED** [review]

**Review by vi-kas — APPROVED** [review]

**Comment by davesmith00000** [comment]
Hello, I was wondering what the status of this was?
I've recently solved all of my cross publishing woes with your wonderful plugin only to discover that SBT 1.0 is in fact a real thing that I need to care about. :-)

**Comment by wyantb-cura** [comment]
Great to see some work on this, this might become a blocker for moving to sbt 1 in one of my repos soon, depending on how sbt-dependency-graph comes along.  Any blockers to this getting merged, such as help testing or anything?

**Comment by ruippeixotog** [comment]
I'm also waiting for this plugin to be released for SBT 1.0. @pauldraper, is there any blocker for this PR to be merged and published?

**Comment by fnmeissner** [comment]
Any update on this?

**Comment by rtimush** [comment]
Hi. Are there any concerns regarding this PR or it is not merged because of not enough time / not needed by Lucid Software / similar reasons?

**Comment by rtimush** [comment]
I just published a scala cross-building extension for [sbt-crossproject](https://github.com/portable-scala/sbt-crossproject), feel free to try: https://github.com/rtimush/sbt-scala-crossproject

**Comment by noamBarkai** [comment]
been using the plugin successfully, now upgrading to sbt 1.X would like to continue using it. I see remarks here 8 months old - has this project stopped being supported / developed?

**Comment by jjudd** [comment]
The project is still support, however the author left Lucid a few months ago. It is being maintained by all of engineering at Lucid, which is not working so well. I will ask internally to see if we can find a specific maintainer.

File: sbt-cross/build.sbt [hunk 1]
```diff
@@ -50,10 +50,12 @@ PgpKeys.pgpPassphrase in Global := Some(Array.emptyCharArray)
 
 sbtPlugin := true
 
-scalacOptions ++= Seq("-deprecation", "-language")
```
Comment (tmccombs) [inline sbt-cross/build.sbt:53]: why did you remove the "-language" option?

Comment (rtimush) [reply sbt-cross/build.sbt:53]: It has been almost a year since I did this, so I don't remember exactly, but probably my rationale was that this option is supposed to be used as `-language:someLanguageFeature`. I'm not sure if specifying just `-language` caused any issues or I just decided that it does nothing and therefore removed it.

**Comment by tmccombs** [comment]
it looks like this needs some more work to get it working. In particular it needs to have crossSbt set to have 0.13 and 1.0, and probably update the sbt-cross project itself to use sbt 1.0 instead of 0.13.

I can work on fixing that when I get more time, or if someone else wants to work on that I'm happy to accept an improved PR.

**Comment by noamBarkai** [comment]
@rtimush - WDYT re latest comments around sbt version? I might try and take a stab at it, though I'm not that experienced with writing sbt plugins.

**Comment by rtimush** [comment]
Since I'm not using this plugin, I'm closing the PR
